### PR TITLE
const var should be replace with IntLiteral

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -427,6 +427,10 @@ export function isArrayType(type: string): boolean {
 	return /^\w[\w.\s{}]+(\[\d+\])+$/.test(type);
 }
 
+// test struct Token {}[N], int[3], st.b.c[N.a]
+export function isConstArrayType(type: string): boolean {
```
Comment (freedomhero) [inline src/utils.ts:431]: no need for this function, just modify reg in `isArrayType` function.

File: src/compilerWrapper.ts [hunk 2]
```diff
@@ -96,6 +97,11 @@ export interface AutoTypedVar {
 	type: string;
 }
 
+export interface IntConstVar {
```
Comment (freedomhero) [inline src/compilerWrapper.ts:100]: maybe `ConstIntVar` for `const int x = 1;`

Comment (zhfnjust) [reply src/compilerWrapper.ts]: removed

File: src/compilerWrapper.ts [hunk 3]
```diff
@@ -478,21 +480,63 @@ function getPublicFunctionDeclaration(mainContract): ABIEntity[] {
...
+	})
+
 	return {
 		contract: mainContract['name'],
 		abi: interfaces
 	};
 }
 
 
+function resolverAbiParamType(type: string, alias: AliasEntity[], staticConstInt: Record<string, number>): string {
```
Comment (freedomhero) [inline src/compilerWrapper.ts:506]: resolveAbiParamType

File: src/compilerWrapper.ts [hunk 4]
```diff
@@ -478,21 +480,63 @@ function getPublicFunctionDeclaration(mainContract): ABIEntity[] {
...
+	} else if(isArrayType(resolvedAliasType)) {
+		const [elemTypeName, arraySizes] = arrayTypeAndSize(resolvedAliasType);
+		let elemType = isStructType(elemTypeName) ? getStructNameByType(elemTypeName) : elemTypeName;
+		return joinArrayTypeAndSize(elemType, arraySizes);
+	}
+
+	return resolvedAliasType;
+}
+
+export function resolverArrayTypeWithConstInt(type: string, staticConstInt: Record<string, number>): string {
```
Comment (freedomhero) [inline src/compilerWrapper.ts:521]: same as above, resolver -> resolve

File: src/utils.ts [hunk 5]
```diff
@@ -482,30 +481,45 @@ export function checkStruct(s: StructEntity, arg: Struct, typeResolver: TypeReso
...
+/**
+ * return eg. int[2][3][4] => ['int', [2,3,4]]
+ * @param arrayTypeName  eg. int[2][3][4]
+ */
+export function arrayTypeAndSize(arrayTypeName: string): [string, Array<number>] {
+  const [elemTypeName, arraySizes] = arrayTypeAndSizeStr(arrayTypeName);
+  return [elemTypeName, arraySizes.map(size => parseInt(size))];
+}
+
+export function joinArrayTypeAndSize(elemTypeName: string, sizes: Array<number>): string {
```
Comment (freedomhero) [inline src/utils.ts:511]: rename to 'toLiteralArrayType'

File: src/utils.ts [hunk 6]
```diff
@@ -424,10 +424,9 @@ export function isStructType(type: string): boolean {
 
 // test struct Token {}[3], int[3], st.b.c[3]
 export function isArrayType(type: string): boolean {
-	return /^\w[\w.\s{}]+(\[\d+\])+$/.test(type);
+	return /^\w[\w.\s{}]+(\[[\w.]+\])+$/.test(type);
```
Comment (freedomhero) [inline src/utils.ts:427]: `.` -> `\.` if you mean literal dot

File: src/utils.ts [hunk 7]
```diff
@@ -482,30 +481,45 @@ export function checkStruct(s: StructEntity, arg: Struct, typeResolver: TypeReso
...
+ * @param arrayTypeName 
  */
-export function arrayTypeAndSize(arrayTypeName: string): [string, Array<number>] {
+ export function arrayTypeAndSizeStr(arrayTypeName: string): [string, Array<string>] {
 
-  const arraySizes: Array<number> = [];
-  [...arrayTypeName.matchAll(/\[([\d])+\]+/g)].map(match => {
-    arraySizes.push(parseInt(match[1]));
+  const arraySizes: Array<string> = [];
+  [...arrayTypeName.matchAll(/\[([\w.]+)\]+/g)].map(match => {
```
Comment (freedomhero) [inline src/utils.ts:492]: same as above, `.` -> `\.`

File: src/utils.ts (hunk 6 above)
Comment (zhfnjust) [reply src/utils.ts:427]: in [] we don't need escape dot

**Pull request merged**